# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/QuotaManager.h
# Commit: 7422fd99c7b1
# Full Hash: 7422fd99c7b16e84aad554bad938366876b30713
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2025-06-13 03:47:27
# Regressor Bug: 1962133
# File Overlap Count: 3
# Overlapping Files: dom/quota/OriginOperations.cpp, dom/quota/ActorsParent.cpp, dom/quota/QuotaManager.h
# Description:
#   Bug 1962133 - QM: Rationalize QuotaManager::CreateDirectoryMetadata2 arguments; r=dom-storage-reviewers,hsingh
#   
#   Refactors CreateDirectoryMetadata2 to take a FullOriginMetadata object as a
#   single argument instead of passing timestamp, persisted and OriginMetadata
#   separately. This simplifies usage, especially in the context of upcoming
# ==============================================================================

diff -r 26405156644d -r 7422fd99c7b1 dom/quota/QuotaManager.h
--- a/dom/quota/QuotaManager.h	Thu Jun 12 17:15:10 2025 +0000
+++ b/dom/quota/QuotaManager.h	Thu Jun 12 17:15:10 2025 +0000
@@ -287,8 +287,7 @@
       const OriginMetadata& aOriginMetadata);
 
   static nsresult CreateDirectoryMetadata2(
-      nsIFile& aDirectory, int64_t aTimestamp, bool aPersisted,
-      const OriginMetadata& aOriginMetadata);
+      nsIFile& aDirectory, const FullOriginMetadata& aFullOriginMetadata);
 
   nsresult RestoreDirectoryMetadata2(nsIFile* aDirectory);
 
